src/include/dhcp-server/widgets.rb

```diff
@@ -27,22 +28,21 @@ def initialize_dhcp_server_widgets(include_target)
       Yast.import "TablePopup"
       Yast.import "SuSEFirewall"
       Yast.import "Mode"
-      Yast.import "DhcpServerUI"
+    end
+
+    # Returns the service for DHCP
+    #
+    # @return [Yast2::SystemService] status service
+    def dhcp_service
+      @service ||= Yast2::SystemService.find(DhcpServer.ServiceName)
     end
 
     # Widget to handle the status of the service
     #
-    # @return [::UI::ServiceStatus] nil if the service is not found (dhcp server
+    # @return [Yast2::ServiceWidget:|nil] nil if the service is not found (dhcp server
     #                               not installed)
     def status_widget
-      return @status_widget unless @status_widget.nil?
-
-      service = DhcpServerUI.service
-      if service
-        @status_widget = ::UI::ServiceStatus.new(service, reload_flag_label: :restart)
-      else
-        nil
-      end
+      @status_widget ||= dhcp_service ? Yast2::ServiceWidget.new(dhcp_service) : nil
```

Comment:
I think we should not silently ignore that service is missing and simply report error. As having DHCP server without service is useless.